packages/financial-templates-lib/src/price-feed/CoinMarketCapPriceFeed.js

```diff
@@ -0,0 +1,168 @@
+    this.convert = convert;
+    this.lookback = lookback;
+    this.networker = networker;
+    this.getTime = getTime;
+    this.minTimeBetweenUpdates = minTimeBetweenUpdates;
+    this.invertPrice = invertPrice;
+    this.priceFeedDecimals = priceFeedDecimals;
+
+    this.priceHistory = []; // array of { time: number, price: BN }
+
+    this.convertPriceFeedDecimals = number => {
+      // Converts price result to wei
+      // returns price conversion to correct decimals as a big number.
+      // Note: Must ensure that `number` has no more decimal places than `priceFeedDecimals`.
+      return this.web3.utils.toBN(
+        parseFixed(number.toString().substring(0, priceFeedDecimals), priceFeedDecimals).toString()
+      );
+    };
+  }
+
+  async update() {
+    const currentTime = this.getTime();
+
+    // Return early if the last call was too recent.
+    if (this.lastUpdateTime !== undefined && this.lastUpdateTime + this.minTimeBetweenUpdates > currentTime) {
+      this.logger.debug({
+        at: "CoinMarketCapPriceFeed",
+        message: "Update skipped because the last one was too recent",
+        currentTime: currentTime,
+        lastUpdateTimestamp: this.lastUpdateTime,
+        timeRemainingUntilUpdate: this.lastUpdateTimes + this.minTimeBetweenUpdates - currentTime
+      });
+      return;
+    }
+
+    this.logger.debug({
+      at: "CoinMarketCapPriceFeed",
+      message: "Updating CoinMarketCapPriceFeed",
+      currentTime: currentTime,
+      lastUpdateTimestamp: this.lastUpdateTime
+    });
+
+    // 1. Construct URL.
+    // See https://coinmarketcap.com/api/documentation/v1/#operation/getV1CryptocurrencyQuotesLatest for how this url is constructed.
+    const url =
+      "https://pro-api.coinmarketcap.com/v1/cryptocurrency/quotes/latest?" +
+      `symbol=${this.symbol}&convert=${this.convert}` +
+      (this.apiKey ? `&CMC_PRO_API_KEY=${this.apiKey}` : "");
+
+    // 2. Send request.
+    const response = await this.networker.getJson(url);
+
+    // 3. Check response.
+    if (!response || !response.data || !response.data[this.symbol] || !response.data[this.symbol].quote[this.convert]) {
+      throw new Error(`🚨Could not parse result from url ${url}: ${JSON.stringify(response)}`);
+    }
+
+    // 4. Parse result.
+    // Return data structure:
+    // {
+    //   "data": {
+    //     "<symbol>": {
+    //       quote: {
+    //         "<convert>": {
+    //           "price": "<price>"
+    //         }
+    //       }
+    //     }
+    //   }
+    // }
+    const newPrice = this.convertPriceFeedDecimals(response.data[this.symbol].quote[this.convert].price);
+
+    // 5. Store results.
+    this.currentPrice = newPrice;
+    this.lastUpdateTime = currentTime;
+    this.priceHistory.push({ time: currentTime, price: newPrice });
+  }
+
+  getCurrentPrice() {
+    return this.invertPrice ? this._invertPriceSafely(this.currentPrice) : this.currentPrice;
+  }
+
+  getHistoricalPrice(time) {
```

Comment:
See same comment about `getHistoricalPrice` that I made for the `CoinGeckoPriceFeed`. Thanks!

Reply:
@nicholaspai took care of this on the latest commit https://github.com/UMAprotocol/protocol/pull/2480/commits/05822407a7026f1fbfac48b2627c935c84961c61